mgexpose/batch_helpers.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+# pylint: disable=R0912,R0914,R0915
+''' Collection of functions to collect MGEs from batches of files'''
+import os
+from collections import Counter, defaultdict
+import json
+
+from dask.distributed import Client, progress, WorkerPlugin
+import dask.bag as db
+from dask.bag import from_delayed
+from dask.delayed import delayed
+import dask
+
+from .gffio import read_mge_genomic_islands_gff
+
+from .base_logger import logger
+import traceback
+
+
+# Create a dictionary: genome: list of MGE IDs. This is needed to filter out only relevant MGEs per genome.The input is a corresponding list (same order) of genomes and MGE IDs.
+def get_genome2mges(genomes, mges):
+    genome2mge_id = {}
+    for id, genome_id in zip(mges, genomes):
+        if genome_id not in genome2mge_id:
+            genome2mge_id[genome_id] = []  # Initialize the list
+        genome2mge_id[genome_id].append(id)
+    return genome2mge_id
+
+# Helper function to extract genome ID/bin ID from file path
+def get_genome_id_from_path(path):
+    """
+    Extract genome ID from a file path.
+    """
+    genome_id = None
+    try:
+        genome_id = path.split("/")[-2]
+    except Exception as e:
+        logger.error(f"Error extracting genome/bin ID{path}: {e}")
+    return genome_id
+
+def collect_batch_mges(gff_paths, i, relevant_ids=None):
+    """
+    Collect MGEs from a batch of GFF files.
+    
+    Parameters:
+    - gff_paths: List of GFF file paths.
+    - i: Index of the batch.
+    - relevant_ids: Optional dictionary of relevant MGE IDs per genome ID.
+
+    Returns:
+    - List of MGE islands for all files in the batch.
+    """
+    islands = []
+    for gff_path in gff_paths:
+        genome_id = get_genome_id_from_path(gff_path)
+        #logger.info(f"Processing genome: {genome_id}")
+
+        try:
+            if relevant_ids:
+                relevant_mges = list(read_mge_genomic_islands_gff(gff_path, relevant_ids[genome_id]))
+            else:
+                relevant_mges = list(read_mge_genomic_islands_gff(gff_path))
+            
+            islands.extend(relevant_mges)
```

Comment:
```suggestion
            if relevant_ids:
                relevant_ids = relevant_ids.get(genome_id)
          
            islands += read_mge_genomic_islands_gff(gff_path, relevant_ids=relevant_ids)
```

1. if relevant_ids is None -> pass through to `read_mge_genomic_islands_gff` function (it's dealt with inside)
2. else, extract the genome-specific ids
3. `+=` is supposedly faster than `.extend()`
4. `read_mge_genomic_islands_gff`, as a generator, does not need to be list-wrapped to be added to `islands`

edit: removed left-over `.extend`